src/transport.c

```diff
@@ -108,10 +108,13 @@ static int find_address(unsigned char ipaddr[BOOTH_IPADDR_LEN],
 				break;
 
 		if (matched == node->addrlen) {
-			*address_bits_matched = matched * 8;
-			*me = node;
-			did_match = EXACT_MATCH;
-			break;
+			if((matched_tmp < matched * 8)||((node->type == SITE)&&(matched_tmp == matched * 8)))
```

Comment:
On Fri, Aug 26, 2016 at 02:56:33AM -0700, Gao,Yan wrote:
> 
> Ah, this explains why I'm not able to produce it with the upstream code;) Thanks for pointing out, Dejan.

OK, good :) I guess that we can close this issue then?